scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,207 @@
+            status : int
+                An integer representing the exit status of the algorithm.
+
+                ``0`` : Optimization terminated successfully.
+
+                ``1`` : Iteration limit reached.
+
+                ``2`` : Problem appears to be infeasible.
+
+                ``3`` : Problem appears to be unbounded.
+
+                ``4`` : Numerical difficulties encountered.
+
+            nit : int
+                The total number of iterations performed in all phases.
+            message : str
+                A string descriptor of the exit status of the algorithm.
+
+    """
+
+    _check_unknown_options(unknown_options)
+
+    statuses = {
+        12: (1, "Iteration limit reached."),
+        9: (0, "Optimization terminated successfully."),
+        8: (3, "The problem is unbounded."),
+        7: (2, "The problem is infeasible."),
+        12: (1, "Iteration limit reached."),
+    }
+    # "Iteration limit reached.",
+    # "The problem appears infeasible, as the phase one auxiliary "
+    # "problem terminated successfully with a residual of {0:.1e}, "
+    # "greater than the tolerance {1} required for the solution to "
+    # "be considered feasible. Consider increasing the tolerance to "
+    # "be greater than {0:.1e}. If this tolerance is unnaceptably "
+    # "large, the problem is likely infeasible.",
+    # "The problem is unbounded, as the simplex algorithm found "
+    # "a basic feasible solution from which there is a direction "
+    # "with negative reduced cost in which all decision variables "
+    # "increase.",
+    # "Numerical difficulties encountered; consider trying "
+    # "method='interior-point'.",
+
+    c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = lp
+
+    lb, ub = bounds.T.copy()                # separate bounds, copy->C-cntgs
+    # highs_wrapper solves LHS <= A*x <= RHS, not equality constraints
+    lhs_ub = -np.ones_like(b_ub)*np.inf     # LHS of UB constraints is -inf
+    rhs_ub = b_ub                           # RHS of UB constraints is b_ub
+    lhs_eq = b_eq                           # Equality constaint is inequality
+    rhs_eq = b_eq                           # constraint with LHS=RHS
+    lhs = np.concatenate((lhs_ub, lhs_eq))
+    rhs = np.concatenate((rhs_ub, rhs_eq))
+
+    if issparse(A_ub) or issparse(A_eq):
+        A = vstack((A_ub, A_eq))
+    else:
+        A = np.vstack((A_ub, A_eq))
+    A = csc_matrix(A)
+
+    options = {
+        'presolve': presolve,
+        'sense': 1,  # minimization
+        'solver': solver,
+        'parallel': parallel,
+        'time_limit': time_limit,
+        'message_level': disp * 1,  # 0 is none, 1 is some. Simplex only?
+        'write_solution_to_file': False,
+        'solution_file': 'test.sol',
+        'write_solution_pretty': True,
+    }
+
+    # this doesn't seem to work, and I don't see an equivalent for IPM
+    if maxiter is not None and solver == 'simplex':
+        options['simplex_iteration_limit'] = maxiter
```

Comment:
Did you find any way to limit iterations for IPM? I don't remember whether it's working for simplex now; if it is we should add a test. If it's not working for either, we'll need to adjust the documentation.

Reply:
- `ipm_iteration_limit` is now available and handled correctly by the wrapper